pkg/controller/kogitotrusty/kogitotrusty_controller_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trusty
+
+import (
+	"testing"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure/services"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	kafkabetav1 "github.com/kiegroup/kogito-cloud-operator/pkg/apis/kafka/v1beta1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/test"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReconcileKogitoTrusty_Reconcile(t *testing.T) {
+	ns := t.Name()
+	instance := &v1alpha1.KogitoTrusty{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-trusty",
+			Namespace: ns,
+		},
+		// We don't need to specify that we need Infinispan, it will figure out that alone :)
+		Spec: v1alpha1.KogitoTrustySpec{},
+	}
+	kafkaList := &kafkabetav1.KafkaList{
+		Items: []kafkabetav1.Kafka{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: "kafka", Namespace: ns},
+				Spec:       kafkabetav1.KafkaSpec{Kafka: kafkabetav1.KafkaClusterSpec{Replicas: 1}},
+				Status: kafkabetav1.KafkaStatus{
+					Listeners: []kafkabetav1.ListenerStatus{
+						{
+							Type:      "plain",
+							Addresses: []kafkabetav1.ListenerAddress{{Host: "kafka", Port: 9092}},
+						},
+					},
+				},
+			},
+		},
+	}
+	cli := test.CreateFakeClient([]runtime.Object{instance, kafkaList}, nil, nil)
+	r := &ReconcileKogitoTrusty{
+		client: cli,
+		scheme: meta.GetRegisteredSchema(),
+	}
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      instance.Name,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	// basic checks
+	res, err := r.Reconcile(req)
```

Comment:
You can reuse the `assertReconcile` function instead: https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/test/reconciler.go#L26

Reply:
Done :+1: